flocker/ca/_script.py

```diff
@@ -90,6 +90,62 @@ def getSynopsis(self):
 
 
 @flocker_standard_options
+class UserCertificateOptions(PrettyOptions):
+    """
+    Command line options for ``flocker-ca create-api-certificate``.
+    """
+
+    helptext = """Create a new certificate for an API end user.
+
+    Creates a certificate signed by a previously generated certificate
+    authority (see flocker-ca initialize command for more information).
+
+    Required parameters:
+
+    * name: A username for which the certificate should be created.
+    """
+
+    synopsis = "<name> [options]"
+
+    optParameters = [
+        ['inputpath', 'i', os.getcwd(),
+         'Path to directory containing root certificate.'],
+        ['outputpath', 'o', os.getcwd(),
+         'Path to directory to write control service certificate.'],
+    ]
+
+    def parseArgs(self, name):
+        self["name"] = name
+
+    def run(self):
+        """
+        Check if root key and certificate files (either default or as
+        specified on the command line) exist in the path and error out if
+        they do not. If there are no path errors, create a new node
+        certificate signed by the root and write it out to the current
+        directory.
+        """
+        self["inputpath"] = FilePath(self["inputpath"])
+        self["outputpath"] = FilePath(self["outputpath"])
+
+        try:
+            try:
+                ca = RootCredential.from_path(self["inputpath"])
+                uc = UserCredential.initialize(
+                    self["outputpath"], ca, self["name"])
+                print (
+                    b"Created {user}.crt. You can now give it to your "
+                    "API enduser so they can access the control service "
+                    "API.".format(user=uc.username)
+                )
```

Comment:
`print` is a really bad thing to use in real programs.  Parameterizing the output stream would make this code much more easily testable and would prevent extra junk from being written to the output stream when running the tests.